package/MDAnalysis/analysis/helix_analysis.py

```diff
@@ -0,0 +1,515 @@
+from .base import AnalysisBase
+
+
+def pdot(a, b):
+    """Pairwise dot product.
+
+    ``a`` must be the same shape as ``b``.
+
+    Parameters
+    ----------
+    a: :class:`numpy.ndarray` of shape (N, M)
+    b: :class:`numpy.ndarray` of shape (N, M)
+
+    Returns
+    -------
+    :class:`numpy.ndarray` of shape (N,)
+    """
+    return np.einsum('ij,ij->i', a, b)
+
+
+def pnorm(a):
+    """Euclidean norm of each vector in a matrix
+
+    Parameters
+    ----------
+    a: :class:`numpy.ndarray` of shape (N, M)
+
+    Returns
+    -------
+    :class:`numpy.ndarray` of shape (N,)
+    """
+    return pdot(a, a)**0.5
+
+
+def vector_of_best_fit(coordinates):
+    """Fit vector through the centered coordinates.
+
+    Parameters
+    ----------
+    coordinates: :class:`numpy.ndarray` of shape (N, 3)
+
+    Returns
+    -------
+    :class:`numpy.ndarray` of shape (3,)
+        Vector of best fit.
+    """
+    centered = coordinates - coordinates.mean(axis=0)
+    Mt_M = np.matmul(centered.T, centered)
+    u, s, vh = np.linalg.linalg.svd(Mt_M)
+    vector = vh[0]
+
+    # does vector face first residue?
+    angle = mdamath.angle(coordinates[0], vector)
+    if angle > np.pi/2 or angle < -np.pi/2:
+        vector *= -1
+    return vector
+
+
+def local_screw(best_fit, ref_axis, rotation_vectors):
+    """
+    Get local screw angles.
+
+    Parameters
+    ----------
+    best_fit: :class:`numpy.ndarray` of shape (3,)
+        Vector of best fit. Tilts are calculated perpendicular to 
+        this axis.
+    ref_axis: :class:`numpy.ndarray` of shape (3,)
+        Reference length-wise axis. One of the reference vectors is 
+        orthogonal to this axis.
+    rotation_vectors: :class:`numpy.ndarray` of shape (N, 3)
+        array of vectors representating the local rotation of each helix.
+
+    Returns
+    -------
+    :class:`numpy.ndarray` of shape (N,)
+        Array of screw angles.
+    """
+    ref_2 = np.cross(ref_axis, best_fit)
+    ref_1 = np.cross(-ref_2, best_fit)  # orthogonal to best_fit and ref_2
+    refs = np.array([ref_1, ref_2])  # (2, 3)
+
+    ref_norms = pnorm(refs)
+    rot_norms = pnorm(rotation_vectors)
+
+    # angles
+    cos = np.matmul(refs, rotation_vectors.T)/np.outer(ref_norms, rot_norms)
+    screw_angles, alt_angles = np.arccos(np.clip(cos, -1, 1))  # (2, n_vec)
+
+    # the effect of the below is to move angles into the (pi/4, 3pi/4 domain)
+    quarter_pi, half_pi = np.pi/4, np.pi/2
+    # replace low angles
+    low = screw_angles < quarter_pi
+    screw_angles[low] = np.where(alt_angles[low] < half_pi,
+                                 half_pi - alt_angles[low],
+                                 alt_angles[low] - half_pi)
+    # replace high angles
+    high = screw_angles > 3*quarter_pi
+    screw_angles[high] = np.where(alt_angles[high] < half_pi,
+                                  half_pi + alt_angles[high],
+                                  3*half_pi - alt_angles[high])
```

Comment:
I may be missing something, but why is this necessary? As far as I can tell, this won't actually result in any changes to any of the angles.

Reply:
Aha, yes, I have no idea what this is meant to achieve, partially because I don't know what a screw angle is meant to describe -- but the original author went to a lot of trouble so I kept it.

> As far as I can tell, this won't actually result in any changes to any of the angles.

I think it does, unless the `local_axis` is exactly in the plane created by `ref_1` and `ref_2` (which depend on `ref_axis` and `global_axes` only) -- this picture is how I understand the angles:
<img width="529" alt="Screenshot 2020-03-20 at 8 55 50 PM" src="https://user-images.githubusercontent.com/31115101/77153365-bc781080-6aed-11ea-820a-8656eb0d30f3.png">

You can see anyway that the screw angles in the comment below are bouncing between 0.5pi-ish, -0.5pi-ish and -pi so my comment `# the effect of the below is to move angles into the (pi/4, 3pi/4 domain)` is clearly wrong.

Reply:
Ah yes, I'd forgotten it wouldn't be exactly in the plane, sorry!

In that case, my assumption is that the screw angle is intended to approximate the cylindrical azimuth angle of each atom about the helix axis, with `ref_1` as 0, but it'll be slightly offset due to the rotation vectors not being quite in plane. Having this code in place would mean we're using the larger of the angles from `ref_1` or the perpendicular `ref_2` to calculate the approx azimuth, which should reduce the relative error we get due to that offset.

Reply:
But we could just calculate the cylindrical azimuth as an angle to the plane of `ref_1` and `global_axes`? Is the approximation a feature?

Reply:
I updated the function to what I think it's trying to achieve: the cylindrical azimuth angle to the plane of the global_axis and ref_axis.